docs/create_allocator_service.md

```diff
@@ -0,0 +1,435 @@
+    ingress.kubernetes.io/https-forwarding-rule: k8s-fws-default-fleet-allocator-ingress--e34f41617775a061
+    ingress.kubernetes.io/https-target-proxy: k8s-tps-default-fleet-allocator-ingress--e34f41617775a061
+    ingress.kubernetes.io/ssl-cert: k8s-ssl-e34f98d0d23b2aca-99c16222402fe1c4--e34f41617775a061
+    ingress.kubernetes.io/url-map: k8s-um-default-fleet-allocator-ingress--e34f41617775a061
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"extensions/v1beta1","kind":"Ingress","metadata":{"annotations":{"kubernetes.io/ingress.allow-http":"false","kubernetes.io/ingress.class":"gce"},"name":"fleet-allocator-ingress","namespace":"default"},"spec":{"backend":{"serviceName":"fleet-allocator-backend","servicePort":8000},"tls":[{"secretName":"allocatorsecret"}]}}
+    kubernetes.io/ingress.allow-http: "false"
+    kubernetes.io/ingress.class: gce
+  creationTimestamp: 2018-09-04T20:00:11Z
+  generation: 1
+  name: fleet-allocator-ingress
+  namespace: default
+  resourceVersion: "4127"
+  selfLink: /apis/extensions/v1beta1/namespaces/default/ingresses/fleet-allocator-ingress
+  uid: 218e0102-b07d-11e8-b86f-42010a8e0072
+spec:
+  backend:
+    serviceName: fleet-allocator-backend
+    servicePort: 8000
+  tls:
+  - secretName: allocatorsecret
+status:
+  loadBalancer:
+    ingress:
+    - ip: 130.211.42.92
+
+```
+
+
+### 9. Check the /healthz Endpoint.
+Inside of examples/allocator-service/main.go, we set up an endpoint /healthz so that kubernetes can determine the health status of the pod.  Let's hit that endpoint now from outside the cluster to see if it returns healthy or not, and to see if the ingress is working.
+
+Since the cert we created earlier is a self signed cert, ignoring insecure warning by adding the -k flag might be necessary:
+```
+curl -k https://the.ip.address/healthz
+```
+
+The output should look like this:
+```
+Healthy
+```
+
+You may need to wait a few moments longer if the output has ssl errors like this:
+```
+curl: (35) error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure
+```
+
+
+### 10. Check Game Servers
+Let's make sure that we have one or more Game Servers in a ready state by running this command:
+```
+kubectl get gs -o=custom-columns=NAME:.metadata.name,STATUS:.status.state,IP:.status.address,PORT:.status.ports
+```
+
+For a fleet of 2 replicas, you should see 2 Game Servers with a Status of Ready:
+```
+NAME                     STATUS    IP              PORT
+simple-udp-thbvz-mqntp   Ready     35.196.249.83   [map[name:default port:7374]]
+simple-udp-thbvz-smchl   Ready     35.229.96.214   [map[name:default port:7016]]
+```
+
+If there is no fleet, please review [Create Game Server Fleet](../docs/create_fleet.md).
+
+
+### 11. Allocate a Game Server
+The service uses Basic Auth to provide some security as to who can allocate gameserver resources.  A generated key is in main.go that you can change.
```

Comment:
Just a thought - why is creating this service better  than just calling the K8s API directly? It has security built in, etc?

Maybe we need a context statement somewhere, that this code would ideally be part of a much larger system, such as a matchmaker.

Reply:
I think the service approach is preferable to calling k8s directly because it creates a separation of concerns, helps protect the game server from being compromised, and abstracts some game server logic from the code that gets deployed with the game client.  Also, being able to refer to a subdomain.domain.ext when creating connections between client and server provides flexibility.

Yes, describing how this service would work as one component in a larger system would be great.  Once horizontal scaling is sorted out, I mean the automatic creation of Ready game servers, it will make more sense.  I will add a sentence about context, and hopefully we can write a separate document in the future that is a complete guide to managing the allocation of gameservers while preserving n gameservers in a ready state.